custom_components/ha-glocaltokens/api.py

```diff
@@ -46,3 +48,48 @@ def get_google_devices_json(self):
     def get_android_id(self):
         """Generate random android_id"""
         return self._client._get_android_id()
+
+    def create_url(self, ip, port, api_endpoint):
+        url = 'https://{ip}:{port}/{endpoint}'.format(
+            ip=ip,
+            port=port,
+            endpoint=api_endpoint
+        )
+        return url
+
+    def get_alarms_and_timers_from(self, ip, port, token, endpoint):
+        url = self.create_url(ip, port, endpoint)
+        HEADERS[HEADER_CAST_LOCAL_AUTH] = token
+        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT) # verify=False is need to avoid SSL security checks. Othervise it will fail to connect
+
+        return response
+
+    def get_google_devices_information(self):
+
+        devices = self._client.get_google_devices_json()
+
+        for device in devices:
+            # To avoid keyerror's
+            device[TIMERS] = []
+            device[ALARMS] = []
+
+            ip = '192.168.0.205'
+            port = str(PORT)
+            token = device[TOKEN]
+            response = self.get_alarms_and_timers_from(ip, port, token, API_ENDPOINT_ALARMS) # IP
+
+            if response.status_code != HTTP_OK:
+                _LOGGER.error("For device {device} - API returned {error}".format(device=device[DEVICE_NAME], error=response.status_code))
+                continue
+
+            result = response.json()
+
+            if TIMERS not in result and ALARMS not in result:
+                _LOGGER.error("For device {device} - {error}".format(device=device[DEVICE_NAME], error=API_RETURNED_UNKNOWN))
+                continue
+
+            device.update(result)
```

Comment:
Sorry for the confusion on this one. If you are already checking that `results` have TIMERS and ALARMS in that, then it means that at this point you will have them available. You might as well split it in this part: `device.update({TIMERS: [...], ALARMS: [...]})`